packages/backend-common/src/discovery/HostDiscovery.ts

```diff
@@ -63,22 +80,67 @@ export class SingleHostDiscovery implements PluginEndpointDiscovery {
 
     const internalBaseUrl = `${protocol}://${host}:${listenPort}`;
 
-    return new SingleHostDiscovery(
+    return new HostDiscovery(
       internalBaseUrl + basePath,
       externalBaseUrl + basePath,
+      config.getOptionalConfig('discovery'),
     );
   }
 
   private constructor(
     private readonly internalBaseUrl: string,
     private readonly externalBaseUrl: string,
+    private readonly discoveryConfig: Config | undefined,
   ) {}
 
+  private getTargetFromConfig(pluginId: string, type: 'internal' | 'external') {
+    const endpoints = this.discoveryConfig?.getOptionalConfigArray('endpoints');
+
+    const target = endpoints
+      ?.find(endpoint => endpoint.getStringArray('plugins').includes(pluginId))
+      ?.get<Target>('target');
+
+    if (!target) {
+      return null;
```

Comment:
just undefined - let's avoid null

Reply:
Addressed in [1364df6](https://github.com/backstage/backstage/pull/17152/commits/1364df62a19fda7c2964c0476fa6d21ad34e7ba0)